Adding configuration files

alexnguyenn opened the pull request:
#120:
Added `.flake8`, `pyproject.toml` (which also contains configuration for black and isort) and `setup.cfg`. 

What still needs to be done:
* Fill out the missing `metadata` fields in `setup.cfg`
* Add `mypy` and `pytest` configurations. 
* Check and correct stuffs (especially dependencies)

alexnguyenn committed:
add .flake8 and pyproject.toml

alexnguyenn committed:
add .flake8 and pyproject.toml

alexnguyenn committed:
add setup.cfg (#120)

alexnguyenn committed:
add setup.cfg (#120)

gwwatkin commented on setup.cfg:
```diff
@@ -0,0 +1,42 @@
+    Topic :: Scientific/Engineering
+    Development Status :: 3 - Alpha
+    Programming Language :: Python :: 3.7
+    Programming Language :: Python :: 3.8
```
Just curious, have you tried running with 3.8 and 3.7. I've only tried 3.9

alexnguyenn replied:
Right now I am using 3.9 but I think I used it with 3.7 and 3.8 before. Might need to check again

gwwatkin commented on setup.cfg:
```diff
@@ -0,0 +1,42 @@
+    mako>=1.1.5
+
+[options.extras_require]
+dev =
```
Guess `mypy` will go here as well

alexnguyenn replied:
Yep, but right now it depends on our progress in [mypy-refactor](https://github.com/latticesurgery-com/lattice-surgery-compiler/tree/mypy-refactor). I don't want to have `mypy` blocking the whole thing.

isolatedinformation committed:
ops.pop() and remove_y_operators() (#135)

* added circuit_has_measurements() to detect Measurements blocks in Circuit
improvments in remove_y_operators() and apply_transformation()

* reverted changes by black by setting line length to 100.
removed unwanted variables and comments

* better calling of circuit_has_improvements()

* minor typo fixes

Co-authored-by: alexnguyenn <hoangminh98@gmail.com>

gwwatkin committed:
Make the source Mypy compliant (#136)

* fix some mypy errors

* more mypy fixes

* Type fixes

* Black formatting

* Apply suggestions from code review

Co-authored-by: Alex Nguyen <46719079+alexnguyenn@users.noreply.github.com>

* Fix not implemented throwing

Co-authored-by: alexnguyenn <hoangminh98@gmail.com>
Co-authored-by: Alex Nguyen <46719079+alexnguyenn@users.noreply.github.com>

alexnguyenn committed:
Merge branch 'add-config-files' of https://github.com/latticesurgery-com/lattice-surgery-compiler into add-config-files

gwwatkin commented:
Is this closed to be rebased on the latest branch?
